src/app/sections/contact-section.tsx

```diff
@@ -0,0 +1,135 @@
+"use client";
+import React, { useState } from "react";
+import { db } from "@/firebase";
+import { collection, addDoc, serverTimestamp } from "firebase/firestore";
+import data from "@/content/home.json";
+export default function ContactSection() {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [message, setMessage] = useState("");
+  const [status, setStatus] = useState<{
+    error: string | null;
+    success: string | null;
+  }>({
+    error: null,
+    success: null,
+  });
+  const [location, setLocation] = useState<{
+    ip: string;
+    city: string;
+    region: string;
+    country: string;
+  } | null>(null);
+
+  const MESSAGES = {
+    REQUIRED_FIELDS: "All fields are required!",
+    SUCCESS: "Message sent successfully!",
+    ERROR: "Failed to send message. Please try again.",
+  };
+
+  const fetchLocation = async () => {
+    if (location) return;
+    const API_KEY = process.env.NEXT_PUBLIC_IP_INFO_TOKEN;
+    const url = `https://ipinfo.io?token=${API_KEY}`;
+
+    try {
+      const response = await fetch(url);
+      const locationData = await response.json();
+      setLocation({
+        ip: locationData.ip,
+        city: locationData.city,
+        region: locationData.region,
+        country: locationData.country,
+      });
+    } catch (error) {
+      console.error("Error fetching location:", error);
+    }
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    const trimmedName = name.trim();
+    const trimmedEmail = email.trim();
+    const trimmedMessage = message.trim();
+
+    if (!trimmedName || !trimmedEmail || !trimmedMessage) {
+      setStatus({
+        error: MESSAGES.REQUIRED_FIELDS,
+        success: null,
+      });
+      setTimeout(() => setStatus({ error: null, success: null }), 5000);
+      return;
+    }
+    await fetchLocation();
+    try {
+      const messagesRef = collection(db, "messages");
+      await addDoc(messagesRef, {
+        name: trimmedName,
+        email: trimmedEmail,
+        message: trimmedMessage,
+        timestamp: serverTimestamp(),
+        location: location,
+      });
+      setStatus({
+        error: null,
+        success: MESSAGES.SUCCESS,
+      });
+      setTimeout(() => setStatus({ error: null, success: null }), 5000);
+      setName("");
+      setEmail("");
+      setMessage("");
+    } catch {
+      setStatus({
+        error: MESSAGES.ERROR,
+        success: null,
+      });
+      setTimeout(() => setStatus({ error: null, success: null }), 5000);
+    }
+  };
+  return (
+    <div className="bg-blue-100 p-8 w-auto h-auto">
+      <div className="text-4xl font-medium text-center mt-8">
+        {data.contact.headline}
+      </div>
+      <form onSubmit={handleSubmit} className="space-y-4">
+        <div className="grid grid-cols-1 md:grid-cols-2 gap-4 mt-13 w-full">
```

Comment:
If the `mt-13` is meant for spacing between the form and the heading. Apply it to the correct element.

```suggestion
      <form onSubmit={handleSubmit} className="space-y-4 mt-13">
        <div className="grid grid-cols-1 md:grid-cols-2 gap-4 w-full">
```